watchtower/blob/justice_kit_test.go

```diff
@@ -212,3 +216,195 @@ func testBlobJusticeKitEncryptDecrypt(t *testing.T, test descriptorTest) {
 			"want: %v, got %v", boj, boj2)
 	}
 }
+
+// TestJusticeKitRemoteWitnessConstruction tests that a JusticeKit returns the
+// proper to-remote witnes script and to-remote witness stack. This should be
+// equivalent to p2wkh spend.
+func TestJusticeKitRemoteWitnessConstruction(t *testing.T) {
+	// Generate the to-remote pubkey.
+	toRemotePrivKey, err := btcec.NewPrivateKey(btcec.S256())
+	if err != nil {
+		t.Fatalf("unable to generate to-remote priv key: %v", err)
+	}
+
+	// Copy the to-remote pubkey into the format expected by our justice
+	// kit.
+	var toRemotePubKey blob.PubKey
+	copy(toRemotePubKey[:], toRemotePrivKey.PubKey().SerializeCompressed())
+
+	// Sign a message using the to-remote private key. The exact message
+	// doesn't matter as we won't be validating the signature's validity.
+	digest := bytes.Repeat([]byte("a"), 32)
+	rawToRemoteSig, err := toRemotePrivKey.Sign(digest)
+	if err != nil {
+		t.Fatalf("unable to generate to-remote signature: %v", err)
+	}
+
+	// Convert the DER-encoded signature into a fixed-size sig.
+	commitToRemoteSig, err := lnwire.NewSigFromSignature(rawToRemoteSig)
+	if err != nil {
+		t.Fatalf("unable to convert raw to-remote signature to "+
+			"Sig: %v", err)
+	}
+
+	// Populate the justice kit fields relevant to the to-remote output.
+	justiceKit := &blob.JusticeKit{
+		CommitToRemotePubKey: toRemotePubKey,
+		CommitToRemoteSig:    commitToRemoteSig,
+	}
+
+	// Now, compute the to-remote witness script returned by the justice
+	// kit.
+	toRemoteScript, err := justiceKit.CommitToRemoteWitnessScript()
+	if err != nil {
+		t.Fatalf("unable to compute to-remote witness script: %v", err)
+	}
+
+	// Assert this is exactly the to-remote, compressed pubkey.
+	if !bytes.Equal(toRemoteScript, toRemotePubKey[:]) {
```

Comment:
Why should the script match the pubkey? The script will have an `OP_0` there along with a data push.

Reply:
this is actually the witness script, i will change the naming

Reply:
To me, this isn't the witness script. Instead, it's the data push of the witness program. Not a blocker, just a matter of tending to consistent language across the codebase.

Reply:
Witness script: the pre-image of the witness program for p2wsh. This is the first thing on the stack. 
Witness program: the pkScript of a segwit output. 
Witness stack: the full witness that allows one to spend a witness program.